src/ring_theory/ideal/basic.lean

```diff
@@ -551,10 +551,7 @@ end comm_ring
 
 namespace ring
 
-variables {R : Type*} [comm_ring R]
-
-lemma not_is_field_of_subsingleton {R : Type*} [ring R] [subsingleton R] : ¬ is_field R :=
-λ ⟨⟨x, y, hxy⟩, _, _⟩, hxy (subsingleton.elim x y)
+variables {R : Type*} [comm_semiring R]
 
 lemma exists_not_is_unit_of_not_is_field [nontrivial R] (hf : ¬ is_field R) :
```

Comment:
Arguably this belongs in a much earlier file (next to `group_with_zero_of_is_unit_or_eq_zero`), but I guess that's maybe out of scope.

A potential golf (after extracting the `let` to `semifield_of_is_unit_or_eq_zero`:
```lean
lemma exists_not_is_unit_of_not_is_field [nontrivial R] (hf : ¬ is_field R) :
  ∃ x ≠ (0 : R), ¬ is_unit x :=
begin
  by_contra,
  apply hf,
  simp_rw [not_exists, not_not, ←or_iff_not_imp_left, or_comm] at h,
  letI : semifield R :=
    { .. group_with_zero_of_is_unit_or_eq_zero h,
      .. ‹comm_semiring R› },
  exact semifield.to_is_field _,
end

Reply:
This proof doesn't rely on anything:
```
import ring_theory.ideal.basic

lemma exists_not_is_unit_of_not_is_field {R} [comm_semiring R] [nontrivial R] (hf : ¬ is_field R) :
  ∃ x ≠ (0 : R), ¬ is_unit x :=
by { simp_rw is_unit_iff_exists_inv, contrapose! hf, exact ⟨exists_pair_ne R, @mul_comm R _, hf⟩ }
```

Reply:
The lemma being discussed is already part of mathlib and not part of this PR, right? So this discussion is not relevant to this PR, right?

Reply:
If you'd said that three months ago I probably had an argument about why it was relevant, but that memory is long since gone. So since I didn't bother to explain why, I guess I agree.

The only slightly messy thing here is that these lemmas are all in the `ring` namespace, but as of this PR are about `comm_semiring`.

So maybe the resolution is just a TODO comment